qiskit/tools/visualization/_circuit_visualization.py

```diff
@@ -521,8 +521,9 @@ def _generate_latex_source(circuit, filename=None,
         str: Latex string appropriate for writing to file.
     """
     dag_circuit = DAGCircuit.fromQuantumCircuit(circuit, expand_gates=False)
-    json_circuit = transpile_dag(dag_circuit, basis_gates=basis, format='json')
-    qcimg = _latex.QCircuitImage(json_circuit, scale, style=style,
```

Comment:
```suggestion
 
    # Deprecate the following lines     -------------------------------------------------|
    if basis is None:                                                                  # |
        basis = ("id,u0,u1,u2,u3,x,y,z,h,s,sdg,t,tdg,rx,ry,rz,"                        # |
                 "cx,cy,cz,ch,crz,cu1,cu3,swap,ccx,cswap")                             # |
    else:                                                                              # |
        warnings.warn('The basis kwarg is deprecated and the circuit drawer '          # |
                      'function will not be able to adjust basis gates itself '        # |
                      'in a future release', DeprecationWarning)                       # |
    basis = basis.split(',') if basis else []                                          # |
    dag = _dagunroller.DagUnroller(dag, _dagbackend.DAGBackend(basis)).expand_gates()  # |
    #     -------------------------------------------------------------------------------|
```

Reply:
Well, @ajavadia said that we shouldn't do this here, he said that he had a plan to retain the basis functionality after #1210 and the other transpiler/dag work he's working on for 0.7. If we add it here (and in the other drawer methods) we'll be adding more places for him to cleanup later. So I'm going to defer to him and we can wait for that work to finish up.

Also, the warning will be duplicated here, we already warn once in circuit_drawer() if basis is set.

Reply:
I removed the warning here: https://github.com/Qiskit/qiskit-terra/pull/1280/files#diff-601179d490e381950db4bf9a4fe01d93R356